crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -2,34 +2,54 @@
 //!
 //! A winning solution becomes a [`Settlement`] once it is executed on-chain.
 
-use crate::{domain::eth, domain::auction::order, infra};
+use crate::{domain::eth, infra};
 
 mod auction;
 mod solution;
 mod transaction;
-pub use {solution::Solution, transaction::Transaction, auction::Auction};
+pub use {auction::Auction, solution::Solution, transaction::Transaction};
 
 /// A solution together with the transaction that executed it on-chain.
 ///
 /// Referenced as a [`Settlement`] in the codebase.
 #[allow(dead_code)]
+#[derive(Debug)]
 pub struct Settlement {
     solution: Solution,
     transaction: Transaction,
+    auction: Auction,
 }
 
 impl Settlement {
-    pub async fn new(tx: eth::TxId, eth: &infra::Ethereum) -> Result<Self, Error> {
+    pub async fn new(
+        tx: eth::TxId,
+        eth: &infra::Ethereum,
+        persistence: &infra::Persistence,
+    ) -> Result<Self, Error> {
         let transaction = eth.transaction(tx).await?;
         let solution = Solution::new(
             &transaction.input.0.clone().into(),
             eth.contracts().settlement_domain_separator(),
         )?;
+        let auction = persistence.get_settlement_auction(&solution).await?;
         Ok(Self {
             solution,
             transaction,
+            auction,
         })
     }
+
+    /// The onchain delivered score of a solution.
+    pub fn score(&self) -> Result<super::competition::Score, solution::error::Score> {
```

Comment:
`promised_score` and `observed_score` or `delivered_score` seem obvious enough to me that one might not even have to read the doc comment.